sage.py

```diff
@@ -215,259 +171,271 @@ def parse(unparsed_data, alert_labels=[], slim=False):
             #    continue
-            count += 1
-            #print(ev[9])
             if SURICATA_SUMMARY:
-                #if cats[ev[6]] != 14:
-                    t[tid][cats[ev[6]]] += 1
+                # if suricata_cats[alert[6]] != 14:
+                percentages[tid][suricata_categories[alert[6]]] += 1
             else:
-                t[tid][ids.index(ev[9])] += 1 
-        #print(count)
-        for i,acat in enumerate(t[tid]):
-            t[tid][i] = acat/len(team)
-        #print('Total percentage: '+ str(sum(t[tid])), 'Actual len: ', str(len(team)))
-    p = []
-    for tid,team in enumerate(unparse):
-        plot = None
+                percentages[tid][micro_attack_stages_codes.index(alert[9])] += 1
+        for i, acat in enumerate(percentages[tid]):
+            percentages[tid][i] = acat / len(team)
+    plots = []
+    for tid, team in enumerate(_team_alerts):
         if tid == 0:
-            plot = plt.bar(ind, t[tid], width)
-        elif tid==1:
-            plot = plt.bar(ind, t[tid], width,
-                 bottom=t[tid-1]) 
+            plot = plt.bar(indices, percentages[tid], bar_width)
+        elif tid == 1:
+            plot = plt.bar(indices, percentages[tid], bar_width, bottom=percentages[tid - 1])
         else:
-            inde = [x for x in range(tid)]
-            bot = np.add(t[0], t[1])
-            for i in inde[2:]:
-                bot = np.add(bot, t[i]).tolist()
-            plot = plt.bar(ind, t[tid], width,
-                 bottom=bot) 
-        p.append(plot)
+            index = [x for x in range(tid)]
+            bottom = np.add(percentages[0], percentages[1])
+            for i in index[2:]:
+                bottom = np.add(bottom, percentages[i]).tolist()
+            plot = plt.bar(indices, percentages[tid], bar_width, bottom=bottom)
+        plots.append(plot)
 
         # TODO: Decide whether to put it like this or normalize over columns
-    #print(t)
-    plt.ylabel('Percentage of occurance')
+    plt.ylabel('Percentage of occurrence')
     plt.title('Frequency of alert category')
     if SURICATA_SUMMARY:
-        plt.xticks(ind, ('c0', 'c1', 'c2', 'c3', 'c4', 'c5', 'c6', 'c7', 'c8', 'c9', 'c10', 'c11', 'c12', 'c13', 'c14'))
+        plt.xticks(indices, ('c0', 'c1', 'c2', 'c3', 'c4', 'c5', 'c6', 'c7', 'c8', 'c9', 'c10', 'c11', 'c12', 'c13', 'c14'))
     else:
-        plt.xticks(ind, [x.split('.')[1] for x in vals], rotation='vertical')
+        plt.xticks(indices, [x.split('.')[1] for x in micro_attack_stages], rotation='vertical')
     plt.tick_params(axis='x', which='major', labelsize=8)
     plt.tick_params(axis='x', which='minor', labelsize=8)
-    #plt.yticks(np.arange(0, 13000, 1000))
-    plt.legend([plot[0] for plot in p], team_labels)
+    # plt.yticks(np.arange(0, 13000, 1000))
+    plt.legend([plot[0] for plot in plots], _team_labels)
     plt.tight_layout()
-    #plt.show() 
-    return plt
+    plt.savefig('data_histogram-' + experiment_name + '.png')
```

Comment:
Instead of returning `plt` and then calling `plt.savefig(...)`, I have moved this call into the method